crates/protocol/src/utils.rs

```diff
@@ -136,3 +141,199 @@ pub fn read_tx_data(r: &mut &[u8]) -> Result<(Vec<u8>, TxType), SpanBatchError>
+            body: alloy_consensus::BlockBody {
+                transactions: vec![op_alloy_consensus::OpTxEnvelope::Deposit(
+                    alloy_primitives::Sealed::new(op_alloy_consensus::TxDeposit {
+                        input: alloy_primitives::Bytes::from(&RAW_BEDROCK_INFO_TX),
+                        ..Default::default()
+                    }),
+                )],
+                ..Default::default()
+            },
+        };
+        let block_hash = block.header.hash_slow();
+        let rollup_config = RollupConfig {
+            genesis: ChainGenesis {
+                l2: BlockNumHash { hash: block_hash, ..Default::default() },
+                ..Default::default()
+            },
+            ..Default::default()
+        };
+        let config = to_system_config(&block, &rollup_config).unwrap();
+        let expected = SystemConfig {
+            batcher_address: address!("6887246668a3b87f54deb3b94ba47a6f63f32985"),
+            overhead: uint!(188_U256),
+            scalar: uint!(684000_U256),
+            gas_limit: 0,
+            base_fee_scalar: None,
+            blob_base_fee_scalar: None,
+            eip1559_denominator: None,
+            eip1559_elasticity: None,
+            operator_fee_scalar: None,
+            operator_fee_constant: None,
+        };
+        assert_eq!(config, expected);
+    }
+
+    #[test]
+    fn test_constructs_isthmus_system_config() {
+        let block = OpBlock {
+            header: alloy_consensus::Header {
+                number: 1,
+                // Holocene EIP1559 parameters stored in the nonce.
+                nonce: hex!("0000beef0000babe").into(),
+                ..Default::default()
+            },
+            body: alloy_consensus::BlockBody {
+                transactions: vec![op_alloy_consensus::OpTxEnvelope::Deposit(
+                    alloy_primitives::Sealed::new(op_alloy_consensus::TxDeposit {
+                        input: alloy_primitives::Bytes::from(&RAW_ISTHMUS_INFO_TX),
+                        ..Default::default()
+                    }),
+                )],
+                ..Default::default()
+            },
+        };
+        let block_hash = block.header.hash_slow();
+        let rollup_config = RollupConfig {
+            genesis: ChainGenesis {
+                l2: BlockNumHash { hash: block_hash, ..Default::default() },
+                ..Default::default()
+            },
+            holocene_time: Some(0),
+            isthmus_time: Some(0),
+            ..Default::default()
+        };
+        assert!(rollup_config.is_holocene_active(block.header.timestamp));
+        let config = to_system_config(&block, &rollup_config).unwrap();
+        let expected = SystemConfig {
+            batcher_address: address!("6887246668a3b87f54deb3b94ba47a6f63f32985"),
+            overhead: U256::ZERO,
+            scalar: uint!(
+                452312848583266388373324160190187140051835877600158453279134670530344387928_U256
+            ),
```

Comment:
I think this is correct since we encode the `blob_base_fee_scalar` and `base_fee_scalar` into the `scalar` field after ecotone.

Reply:
Yeah looks correct. In hex this is `0x10000000000000000000000000000000000000000000000000c5fc500000558`, which looks like the proper encoding (w/ the version byte.)